assignments/assignment04.rb

```diff
@@ -12,7 +12,13 @@
 # F[n] -> F[n-2] + F[n-1]
 
 def fib(n)
-  # your implementation here
+  def fib_helper(n, acc1, acc2)
```

Comment:
This is an interesting implementation. I seriously doubt that you figured this out by yourself. This is really taking an iterative solution, and then converting it back to a recursive implementation of that iterative solution. Nowhere in this solution do you ever see the `fib(n-1) + fib(n-2)` from the above definition.

If you truly understand this solution and why it works, great. If you don't, I highly recommend that you don't just google for solutions to problems and then copy the code that you see without understanding it.